packages/Processor.ts

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env node
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Imperative } from "@zowe/imperative";
+import * as net from "net";
+import { DaemonClient } from "./DaemonClient";
+
+// TODO(Kelosky): handle prompting cases from login command
+// TODO(Kelosky): prompt* broken - hangs, must restart daemon
+// TODO(Kelosky): colors do not come through on some terminals (must be started via `node lib/main --daemon` to see colors)
+// TODO(Kelosky): stderr
+// TODO(Kelosky): plugins install
+
+/**
+ * Initial paramter parse to handle conditionally starting as a persistent process (e.g. daemon mode)
+ * @export
+ * @class Processor
+ */
+export class Processor {
+
+    /**
+     * Default port number
+     * @private
+     * @static
+     * @memberof Processor
+     */
+    private static readonly DEFAULT_PORT = 4000;
+
+    /**
+     * Undocumented paramter for launching in server mode
+     * @private
+     * @static
+     * @memberof Processor
+     */
+    private static readonly DAEMON_KEY = "--daemon";
+
+    /**
+     * Undocumented paramter for launching in server mode with a port
+     * @private
+     * @static
+     * @memberof Processor
+     */
+    private static readonly DAEMON_PORT_KEY = "--daemon=";
```

Comment:
if this is intended to configure the port in which the daemon instance is listening to, should it be renamed to something like `--daemon-port=`